src/txn/txn_ckpt.c

```diff
@@ -15,6 +15,153 @@ static int __checkpoint_presync(WT_SESSION_IMPL *, const char *[]);
+ */
+static int
+__checkpoint_flush_tier_wait(WT_SESSION_IMPL *session, const char **cfg)
+{
+    WT_CONFIG_ITEM cval;
+    WT_CONNECTION_IMPL *conn;
+    uint64_t now, start, timeout;
+    int yield_count;
+
+    conn = S2C(session);
+    yield_count = 0;
+    now = start = 0;
+    /*
+     * The internal thread needs the schema lock to perform its operations and flush tier also
+     * acquires the schema lock. We cannot be waiting in this function while holding that lock or no
+     * work will get done.
+     */
+    WT_ASSERT(session, !FLD_ISSET(session->lock_flags, WT_SESSION_LOCKED_SCHEMA));
+    WT_RET(__wt_config_gets(session, cfg, "flush_tier.timeout", &cval));
+    timeout = (uint64_t)cval.val;
+    if (timeout != 0)
+        __wt_seconds(session, &start);
+
+    /*
+     * It may be worthwhile looking at the add and decrement values and make choices of whether to
+     * yield or wait based on how much of the workload has been performed. Flushing operations could
+     * take a long time so yielding may not be effective.
+     */
+    while (!WT_FLUSH_STATE_DONE(conn->flush_state)) {
+        if (start != 0) {
+            __wt_seconds(session, &now);
+            if (now - start > timeout)
+                return (EBUSY);
+        }
+        if (++yield_count < WT_THOUSAND)
+            __wt_yield();
+        else {
+            __wt_cond_signal(session, conn->tiered_cond);
+            __wt_cond_wait(session, conn->flush_cond, 200, NULL);
+        }
+    }
+    return (0);
+}
+
+/*
+ * __checkpoint_flush_tier --
+ *     Perform one iteration of tiered storage maintenance.
+ */
+static int
+__checkpoint_flush_tier(WT_SESSION_IMPL *session, bool force)
+{
+    WT_CKPT ckpt;
+    WT_CONFIG_ITEM cval;
+    WT_CONNECTION_IMPL *conn;
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    uint64_t ckpt_time, flush_time;
+    const char *key, *value;
+
+    __wt_verbose(session, WT_VERB_TIERED, "CKPT_FLUSH_TIER: Called force %d", force);
+
+    WT_STAT_CONN_INCR(session, flush_tier);
+    conn = S2C(session);
+    cursor = NULL;
+    /*
+     * For supporting splits and merge:
+     * - See if there is any merging work to do to prepare and create an object that is
+     *   suitable for placing onto tiered storage.
+     * - Do the work to create said objects.
+     * - Move the objects.
+     */
+    conn->flush_state = 0;
+
+    /*
+     * We hold the checkpoint lock so we know no other thread can be doing a checkpoint at this time
+     * but our time can move backward with respect to the time set by a different thread that did a
+     * checkpoint. Update time value for most recent flush_tier, taking the more recent of now or
+     * the checkpoint time.
+     */
+    WT_ASSERT(session, FLD_ISSET(session->lock_flags, WT_SESSION_LOCKED_CHECKPOINT));
+    __wt_seconds(session, &flush_time);
+    /*
+     * XXX If/when flush tier no longer requires the checkpoint lock, all of these global values and
```

Comment:
Since we're in checkpoint and have the checkpoint lock, this comment doesn't make sense to me. Maybe just change this to a reminder that we have the checkpoint lock so the following is safe?  Or remove the comment if you think it is obvious.